platforms/enterprise/enterprise-plugin-performance/build.gradle.kts

```diff
@@ -59,11 +59,22 @@ subprojects {
 }
 
 tasks.withType<gradlebuild.performance.tasks.PerformanceTest>().configureEach {
-    systemProperties["incomingArtifactDir"] = "$rootDir/incoming/"
-
     environment("GRADLE_INTERNAL_REPO_URL", System.getenv("GRADLE_INTERNAL_REPO_URL"))
     environment("GRADLE_INTERNAL_REPO_USERNAME", System.getenv("GRADLE_INTERNAL_REPO_USERNAME"))
     environment("GRADLE_INTERNAL_REPO_PASSWORD", System.getenv("GRADLE_INTERNAL_REPO_PASSWORD"))
 
     reportGeneratorClass = "org.gradle.performance.results.BuildScanReportGenerator"
+
+    val projectRootDir = project.rootDir
+    val pluginInfoDir = project.providers.gradleProperty("enterprisePluginInfoDir")
+        .orElse(projectRootDir.resolve("incoming").path)
+        .map { projectRootDir.resolve(it) }
+
+    // System property required by `AbstractBuildScanPluginPerformanceTest`
+    jvmArgumentProviders += object : CommandLineArgumentProvider {
```

Comment:
🤔 It might be easier to read if you use an actual class instead of using an anonymous class here.

Reply:
I would make it easier to read, and I don't like this fact. Somehow, it feels wrong that a (BT) user needs to introduce a class (and come up with a name for it!) for a one-off thing.

Reply:
Changed it to a class